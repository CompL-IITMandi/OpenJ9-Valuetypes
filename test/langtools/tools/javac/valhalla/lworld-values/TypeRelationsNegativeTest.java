/*
 * @test /nodynamiccopyright/
 * @bug 8222792
 * @summary Javac should enforce the latest relationship rules between an inline type and its nullable projection
 * @compile/fail/ref=TypeRelationsNegativeTest.out -XDrawDiagnostics TypeRelationsNegativeTest.java
 */

final value class TypeRelationsNegativeTest {

    void foo() {
        TypeRelationsNegativeTest x = null; // error
        TypeRelationsNegativeTest? xq = null;

        xq = x;
        xq = (TypeRelationsNegativeTest?) x;
        xq = (TypeRelationsNegativeTest) x;
        x = xq;  // error
        x = (TypeRelationsNegativeTest?) xq; // error
        x = (TypeRelationsNegativeTest) xq;
 
        TypeRelationsNegativeTest [] xa = new TypeRelationsNegativeTest[] { null }; // error
        TypeRelationsNegativeTest? [] xqa = new TypeRelationsNegativeTest?[] { null };

        xqa = xa;
        xqa = (TypeRelationsNegativeTest?[]) xa;
        xa = xqa; // error
        xa = (TypeRelationsNegativeTest []) xqa;
    }
    int x = 10;
}
