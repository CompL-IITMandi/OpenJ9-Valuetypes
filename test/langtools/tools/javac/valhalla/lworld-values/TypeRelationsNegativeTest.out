TypeRelationsNegativeTest.java:11:39: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: compiler.misc.type.null, TypeRelationsNegativeTest)
TypeRelationsNegativeTest.java:17:13: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: TypeRelationsNegativeTest?, TypeRelationsNegativeTest)
TypeRelationsNegativeTest.java:18:13: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: TypeRelationsNegativeTest?, TypeRelationsNegativeTest)
TypeRelationsNegativeTest.java:21:77: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: compiler.misc.type.null, TypeRelationsNegativeTest)
TypeRelationsNegativeTest.java:26:14: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: TypeRelationsNegativeTest?[], TypeRelationsNegativeTest[])
5 errors
